### DISCUSSION TITLE
请问如何实现按序登录？

### DISCUSSION THREAD

**Discussion by Arondight** [takayama-lily/oicq#290] [Q&A] [answered]
@takayama-lily 我现在有若干个 createClient 返回值的数组，对其中每一个元素监听二维码事件并尝试登录，请问如何做到一个丘丘登录完毕（成功或失败）后再开始下一个丘丘的登录？

**Comment by takayama-lily** [comment, marked as answer]
包装成Promise

**Reply by Arondight** [reply]
是 client.on 的第二个参数传递一个 promise 进去吗

**Reply by takayama-lily** [reply]
不是。把你期待的异步结果变成一个Promise，等待他结束。

**Reply by Arondight** [reply]
这样必须在一个 promise 中完成 createClient 和 client.on 
吗？

**Reply by Arondight** [reply]
可以了，感谢老大哥
